quilt/tools/build.py

```diff
@@ -135,38 +164,60 @@ def _callback(count):
         failover_args = {}
         failover_args.update(failover)
         failover_args.update(kwargs)
-        df = handler(path, **failover_args)
+        dataframe = handler(path, **failover_args)
 
     # cast object columns to strings
-    for name, col in df.iteritems():
+    for name, col in dataframe.iteritems():
         if col.dtype == 'object':
-            df[name] = col.astype(str)
+            dataframe[name] = col.astype(str)
 
-    return df
+    return dataframe
 
-def build_package(username, package, yaml_path):
+def build_package(username, package, yaml_path, checks_path=None, dry_run=False, env='default'):
     """
     Builds a package from a given Yaml file and installs it locally.
 
     Returns the name of the package.
     """
+    def find(key, value):
+        """find all nodes transitively"""
+        for k, v in value.items():
+            if k == key:
+                yield v
+            elif isinstance(v, dict):
+                for result in find(key, v):
+                    yield result
+            elif isinstance(v, list):
+                for item in v:
+                    for result in find(key, item):
+                        yield result
     build_dir = os.path.dirname(yaml_path)
     fd = open(yaml_path)
     docs = yaml.load_all(fd)
-    data = next(docs, None) # leave other dicts in the generator
-    if not isinstance(data, dict):
-        raise BuildException("Unable to parse YAML: %s" % yaml_path)
-
-    build_package_from_contents(username, package, build_dir, data)
-
-def build_package_from_contents(username, package, build_dir, data):
-    """
-    Builds package according to YAML node
-    """
-    contents = data.get('contents', {})
+    build_data = next(docs, None) # leave other dicts in the generator
+    if not isinstance(build_data, dict):
+        raise BuildException("Unable to parse build YAML: %s" % yaml_path)
+    # in-lined checks (in build.yml) take precedence over checks.yml
+    if find('checks', build_data['contents']) and 'checks' not in build_data:
```

Comment:
Also, I think you do need to call `next(find(...))`. Even an empty generator evaluates to `True`.

Reply:
DONE (as part of rewrite)